rules/falco_rules.yaml

```diff
@@ -3120,6 +3120,17 @@
   priority: NOTICE
   tags: [network, process, mitre_command_and_control]
 
+# This rule helps detect CVE-2021-4034:
+# A privilege escalation to root through memory corruption
+- rule: Polkit Local Privilege Escalation Vulnerability (CVE-2021-4034)
+  desc: "This rule detects an attempt to exploit a privilege escalation vulnerability in Polkit's pkexec. By running specially crafted code, a local user can leverage this flaw to gain root privileges on a compromised system"
+  condition:
+    spawned_process and user.uid != 0 and proc.name=pkexec and proc.args = '' and proc.env icontains 'GCONV_PATH'
```

Comment:
Why `GCONV_PATH` and not just `PATH`? The way the original vulnerability was described (https://blog.qualys.com/vulnerabilities-threat-research/2022/01/25/pwnkit-local-privilege-escalation-vulnerability-discovered-in-polkits-pkexec-cve-2021-4034) it was a manipulation of `PATH`. I wonder if we even need to check `proc.env` at all; it seems that any invocation of `pkexec` with an empty set of args would be suspect, no? Or am I misunderstanding what this all means to falco?

Reply:
Hey @jhoos, thanks for the feedback. 
I added this condition due to the exploitability and GCONV_PATH can force iconv_open() to read another configuration file. Ee-introducing GCONV_PATH into pkexec's environment, and to execute our own shared library it's possible to get root. and exploit CVE-2021-4034.
Reference: https://www.qualys.com/2022/01/25/cve-2021-4034/pwnkit.txt

Reply:
Jason, I tend to agree with you here. The exploit relies on a malformed environment variables which is hard to detect during the process execution (no baseline could compare). And execute `pkexe` may be a normal case which may give you bunch of options (the same behavior as `-h` or `--help`). If what I assumed before is correct, I don't see there is an easy to write a rule to detect such exploit.

Reply:
Thanks @Kaizhe for your feedback. I changed the condition removing the proc.env check.
Thanks again.